File: portfolio/Seed/SeedAdmin.cs
using System.Text;
using Dapper;
using Npgsql;
using portfolio.Auth;
using portfolio.Models;

namespace portfolio.Seed;
public class SeedAdmin
{
    public static async Task Seed(string connectionString, IConfiguration configuration)
    {
        Admin[] adminData = AdminData.GetAdminData();

        // connect to database with dapper
        await using var connection = new NpgsqlConnection(connectionString);

        Console.WriteLine("Seeding admin...");
        Console.WriteLine("--------------------------------------------------------------");

        // hash password
        foreach (Admin admin in adminData)
        {
            admin.Password = new AuthService(configuration).HashPassword(admin.Password!);
        }

        // seed admin table
        foreach (Admin admin in adminData)
        {
            StringBuilder sql = new StringBuilder();
            sql.Append("INSERT INTO admin (username, password, role) VALUES (");
            sql.Append("@Username, @Password, @Role");
            sql.Append(")");
            await connection.ExecuteAsync(sql.ToString(),
                new
                {
                    Username = admin.Username,
                    Password = admin.Password,
                    Role = admin.Role,
                }
            );
        }

        Console.WriteLine("Seeding complete!");
        Console.WriteLine("--------------------------------------------------------------");
    }
}

File: extractSeedDb/seedAdmin.js
const { pool } = require('./dbConnection.js');

async function seedAdmin() {
  try {
    const adminData = await fs.promises.readFile('SeedData/admin_data.json', 'utf8');
    const adminInfo = JSON.parse(adminData);

    await pool.query('DELETE FROM admin');

    for (const info of adminInfo) {
      const query = `INSERT INTO admin
                     (username, password, role)
                     VALUES ($1, $2, $3)`;
      const values = [info.name, info.password, info.role];

      await pool.query(query, values);
    }

    pool.end();
  } catch (err) {
    console.error('Error seeding database:', err);
  }
}
